### PR TITLE
[bugfix](backup)(cooldown) cancel backup properly when be backup failed

### PR DESCRIPTION
**Pull request by justfortaste**
Currently, when a backup job failed, but it still at the state of SNAPSHOTING.

Cancel the cancel backup properly when be backup failed

### CODE REVIEW

**Comment by nextdreamblue** [comment]
run buildall

**Review by xy720 — DISMISSED** [review]
LGTM

**Comment by xy720** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/backup/BackupJob.java [hunk 1]
```diff
@@ -169,6 +169,55 @@ public BackupContent getContent() {
...
+    private synchronized boolean tryNewTabletSnapshotTask(SnapshotTask task) {
+        Table table = env.getInternalCatalog().getTableByTableId(task.getTableId());
+        if (table == null) {
+            return false;
+        }
+        OlapTable tbl = (OlapTable) table;
+        if (tbl.getId() != task.getTableId()) {
+            return false;
+        }
+        Partition partition = tbl.getPartition(task.getPartitionId());
```
Comment (w41ter) [inline fe/fe-core/src/main/java/org/apache/doris/backup/BackupJob.java:181]: Here we need to add a read lock for OlapTable

File: fe/fe-core/src/main/java/org/apache/doris/backup/BackupJob.java [hunk 2]
```diff
@@ -181,6 +230,21 @@ public synchronized boolean finishTabletSnapshotTask(SnapshotTask task, TFinishT
...
                 cancelInternal();
             }
+
+            if (request.getTaskStatus().getStatusCode() == TStatusCode.TABLET_MISSING) {
+                if (!tryNewTabletSnapshotTask(task)) {
+                    status = new Status(ErrCode.NOT_FOUND,
+                            "make snapshot failed, failed to ge tablet, table will be droped or truncated");
+                    cancelInternal();
+                }
+            }
```
Comment (w41ter) [inline fe/fe-core/src/main/java/org/apache/doris/backup/BackupJob.java:240]:
```suggestion
            if (request.getTaskStatus().getStatusCode() == TStatusCode.TABLET_MISSING
            		&& !tryNewTabletSnapshotTask(task)) {
                  status = new Status(ErrCode.NOT_FOUND,
                          "make snapshot failed, failed to ge tablet, table will be droped or truncated");
                  cancelInternal();
            }
```

**Review by w41ter — COMMENTED** [review]
Reset LGTM.

File: regression-test/suites/backup_restore/test_backup_cancelled.groovy [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+        values.add("(${i}, ${i})")
+    }
+    sql "INSERT INTO ${dbName}.${tableName} VALUES ${values.join(",")}"
+    def result = sql "SELECT * FROM ${dbName}.${tableName}"
+    assertEquals(result.size(), values.size());
+
+
+    // test failed to get tablet when truncate or drop table
+
+    GetDebugPoint().enableDebugPointForAllBEs("SnapshotManager::make_snapshot.inject_failure", [execute:3]);
```
Comment (w41ter) [inline regression-test/suites/backup_restore/test_backup_cancelled.groovy:54]: It seems that this injection will affect other cases running in parallel

File: fe/fe-core/src/main/java/org/apache/doris/backup/BackupJob.java (hunk 1 above)
Comment (justfortaste) [reply fe/fe-core/src/main/java/org/apache/doris/backup/BackupJob.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/backup/BackupJob.java (hunk 2 above)
Comment (justfortaste) [reply fe/fe-core/src/main/java/org/apache/doris/backup/BackupJob.java]: done

File: regression-test/suites/backup_restore/test_backup_cancelled.groovy (hunk 3 above)
Comment (justfortaste) [reply regression-test/suites/backup_restore/test_backup_cancelled.groovy]: ok

**Comment by dataroaring** [comment]
run buildall

**Review by w41ter — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Comment by dataroaring** [comment]
run buildall

**Comment by nextdreamblue** [comment]
run cloud_p1

**Pull request merged**